js/plates-model/plate.js

```diff
@@ -85,12 +85,38 @@ export default class Plate extends PlateBase {
   }
 
   updateCenter () {
+    if (this.center) {
+      const centerField = this.fieldAtAbsolutePos(this.center)
+      // Don't bother moving the label if it's still on a visible field
+      if (centerField && !centerField.colliding) {
+        return
+      }
+    }
+
+    // If the current label position is no longer valid, move it back to the center
     let sum = new THREE.Vector3(0, 0, 0)
     this.fields.forEach(field => {
-      sum = sum.add(field.absolutePos)
+      if (!field.colliding) {
```

Comment:
@pjanik I struggled a lot trying to figure out how to only include "visible" fields in the calculations. I tried using `field.subduction`, but I think that missed some fields that were still under other plates? I ended up using `field.colliding`, but it's not perfect, because there are some visible fields colliding with others and skipping them throws off the calculations.